tests/src/test/java/com/maxio/advancedbilling/controllers/proformainvoices/ProformaInvoicesControllerSubscriptionGroupsTest.java

```diff
@@ -0,0 +1,210 @@
+
+/**
+ * This class tests both `Create Consolidated Proforma Invoices` and `List Subscription Group Proforma Invoices`
+ * operations, as the Creation response is asynchronous and has no specific response, but it's needed to execute listing.
+ */
+public class ProformaInvoicesControllerSubscriptionGroupsTest {
+
+    private static final TestSetup TEST_SETUP = new TestSetup();
+    private static final AdvancedBillingClient CLIENT = TestClient.createClient();
+    private static final ProformaInvoicesController PROFORMA_INVOICES_CONTROLLER = CLIENT
+            .getProformaInvoicesController();
+
+    private static SubscriptionGroupSignupResponse groupSignup;
+    private static ProductFamily productFamily;
+    private static Product product;
+
+    @BeforeAll
+    static void setUp() throws IOException, ApiException, InterruptedException {
+        productFamily = TEST_SETUP.createProductFamily();
+        product = TEST_SETUP.createProduct(productFamily, b -> b.priceInCents(1250));
+        Component meteredComponent = TEST_SETUP.createMeteredComponent(productFamily, 11.5);
+
+        groupSignup = TEST_SETUP.signupWithSubscriptionGroup(product, meteredComponent);
+        PROFORMA_INVOICES_CONTROLLER.createConsolidatedProformaInvoice(groupSignup.getUid());
+
+        Thread.sleep(5000);
+    }
+
+    @AfterAll
+    static void teardown() throws IOException, ApiException {
+        TestTeardown testTeardown = new TestTeardown();
+        testTeardown.deleteSubscriptionGroup(groupSignup);
+    }
+
+    @Test
+    void shouldListGroupProformas() throws IOException, ApiException {
+        // given-when
+        ListProformaInvoicesResponse listProformaInvoicesResponse = PROFORMA_INVOICES_CONTROLLER
+                .listSubscriptionGroupProformaInvoices(new ListSubscriptionGroupProformaInvoicesInput.Builder()
+                        .uid(groupSignup.getUid())
+                        .lineItems(true)
+                        .build()
+                );
+
+        // then
+        assertThat(listProformaInvoicesResponse.getMeta()).isNotNull();
+        assertThat(listProformaInvoicesResponse.getMeta().getCurrentPage()).isEqualTo(1);
+        assertThat(listProformaInvoicesResponse.getMeta().getStatusCode()).isEqualTo(200);
+        assertThat(listProformaInvoicesResponse.getMeta().getTotalCount()).isEqualTo(1);
+        assertThat(listProformaInvoicesResponse.getMeta().getTotalPages()).isEqualTo(1);
+
+        assertThat(listProformaInvoicesResponse.getProformaInvoices().size()).isEqualTo(1);
+
+        ProformaInvoice proformaInvoice = listProformaInvoicesResponse.getProformaInvoices().get(0);
+
+        InvoiceAddress invoiceBillingAddress = proformaInvoice.getBillingAddress();
+        assertThat(invoiceBillingAddress.getAdditionalProperties()).isEmpty();
+        assertAll(
+                () -> assertThat(invoiceBillingAddress).isNotNull(),
+                () -> assertThat(invoiceBillingAddress.getStreet()).isNull(),
```

Comment:
Why address is null? Shouldn't it use customers address?

Reply:
customer address (subscription group signup `payer attributes`) is part of `invoiceShippingAddress`, invoiceBillingAddress contain address specified in subscription group signup `creditCardAttributes`, these properties are empty in the shared method that creates a group. I'll fill them with some data and correct assertions